src/libexpr/nixexpr.hh

```diff
@@ -79,16 +79,28 @@ struct Expr
 {
     virtual ~Expr() { };
     virtual void show(std::ostream & str) const;
+    virtual void showAsAterm(std::ostream & str) const;
+    virtual void showAsJson(std::ostream & str) const;
     virtual void bindVars(const StaticEnv & env);
     virtual void eval(EvalState & state, Env & env, Value & v);
     virtual Value * maybeThunk(EvalState & state, Env & env);
     virtual void setName(Symbol & name);
 };
 
+struct ExprAsAterm : Expr
+{
+    void show(std::ostream & str) const;
+};
+
+struct ExprAsJson : Expr
+{
+    void show(std::ostream & str) const;
+};
+
 std::ostream & operator << (std::ostream & str, const Expr & e);
 
 #define COMMON_METHODS \
-    void show(std::ostream & str) const; \
+    void showAsAterm(std::ostream & str) const; \
```

Comment:
Let's not rename this to minimize code churn.

Reply:
done